### DISCUSSION TITLE
Table widget in Edit page

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#6600] [Help] [answered]
Hello everyone, 

How can I create a table (widget) in Edit page? I create a widget using php artisan make:filament-widget ExampleWidget, but didn't find anything how I can show widget details in table.

**Comment by danharrin** [comment]
https://filamentphp.com/docs/2.x/admin/resources/widgets#displaying-a-widget-on-a-resource-page

**Reply by ghost** [reply]
Thank you, I already display it in Card. But want to display it in a table with pagination, filter and search. How can I do it?

**Reply by danharrin** [reply]
I don't know what you mean, can you screenshot?

**Reply by ghost** [reply]
https://prnt.sc/Y6lG449nF3ut this table I create with HTML code. there is no pagination, no filter function. I mean just simple html table. I need to pass database values to filamentphp table with all benefits (like filter, pagination and so on). This table on Edit page.

**Reply by danharrin** [reply, marked as answer]
You can use the table builder to create tables with filters and pagination. Check the docs "table widget" and the docs "tables - getting started"

**Reply by ghost** [reply]
```
public ?ModelFirst $record = null;

public function record(ModelFirst $record)
{
    $this->record = $record;
    return $this;
}

protected function getTableQuery(): Builder
{
    $id = $this->record->log_id;
    $query = ModelSecond::query();
    return $query->where('log_id', $id);
}

protected function getTableColumns(): array
{
    return [
        Tables\Columns\TextColumn::make('Result')->default(function(){
            $json = json_decode($this->result, true);
            foreach ($json as $value) {
                return $value['message'];
            }
        }),
    ];
}
```
@danharrin Thank you for your answer.

This is widget part of code, I'm trying to get 'log_id' from ModelFirst and pass it to ModelSecond to get rows from ModelSecond (this part is ok). 

When I try to show it on table with `Tables\Columns\TextColumn::make('result')` it return JSON (as value in json, it's ok). 

Then I try to get values from this JSON by using **default()**, but I'm getting error **Property [$result] not found on component: [app.filament.resources.....-widget]**

P.S.: in this code as you see I use `Tables\Columns\TextColumn::make('Result')`, because if I use 'result' it will return me json.

**Reply by ghost** [reply]
I fix it like this

```
Tables\Columns\TextColumn::make('Name')
    ->default(function(Model $record){
        $json = array($record->result);
        foreach ($json as $key => $value) {
            $valueJson = json_decode($value, true);
            $data = [
                'chat_id' => $valueJson['value1'],
                'message' => $valueJson['value2'],
                'statusCode' => $valueJson['value3'],
                'datetime' => $record->created_at,
            ];
            return view('my-blade')->with($data);
        }
    }),
```
